docs/docsite/rst/network/getting_started/network_differences.rst

```diff
@@ -0,0 +1,72 @@
+
+Multiple Communication Protocols
+================================================================================
+
+Because network modules execute on the control node instead of on the managed nodes, they can support multiple communication protocols. The communication protocol (XML over SSH, CLI over SSH, API over HTTPS) selected for each network module depends on the platform and the purpose of the module. Some network modules support only one protocol; some offer a choice. The most common protocol is CLI over SSH. You set the communication protocol with the ``ansible_connection`` variable:
+
+.. csv-table::
+   :header: "Value of ansible_connection", "Protocol", "Requires"
+   :widths: 30, 10, 10
+
+   "network_cli", "CLI over SSH", "network_os setting"
+   "netconf", "XML over SSH", "network_os setting"
+   "local", "depends on provider", "provider setting"
+
+Beginning with Ansible 2.5, we recommend using ``network_cli`` or ``netconf`` for ``ansible_connection`` whenever possible. For details on using API over HTTPS connections, see the platform-specific pages.
+
+Modules Organized by Network Platform
+================================================================================
+
+A network platform is a set of network devices with a common operating system that can be managed by a collection of modules.  The modules for each network platform share a prefix, for example: 
+
+- Arista: ``eos_``
+- Cisco: ``ios_``, ``iosxr_``, ``nxos_``
+- Juniper: ``junos_``
+- VyOS ``vyos_``
+
+All modules within a network platform share certain requirements. Some network platforms have specific differences - see the platform-specific documentation for details.
+
+Privilege Escalation: `authorize` and `become`
+================================================================================
+
+Several network platforms support privilege escalation, where certain tasks must be done by a privileged user. This is generally known as ``enable`` mode (the equivalent of ``sudo`` in \*nix administration). Ansible network modules offer privilege escalation for those network devices that support it. However, different platforms use privilege escalation in different ways. 
+
+Network platforms that support ``connection: network_cli`` and privilege escalation use the Ansible parameter ``become: yes`` with ``become_method: enable``. For modules in these platforms, a ``group_vars`` file would look like:
+
+.. code-block:: yaml
+
+   ansible_connection: network_cli
+   ansible_network_os: ios
+   ansible_become: yes
+   ansible_become_method: enable
+
+We recommend using ``network_cli`` connections whenever possible. However, in Ansible < 2.5, some network platforms support privilege escalation but not ``network_cli`` connections. In that case, you must use ``authorize: yes`` and ``auth_pass: my_enable_password`` in the ``provider`` settings. For that use case, a ``group_vars`` file would look like:
```

Comment:
Well privilege escalation needs to in provider for version < 2.5.
But `provider` is still required in 2.5 until we implement nxapi and eapi connection plugin in 2.6. So this is not the only reason we use provider. 
may be we rephrase this to:
`In an effort to help streamline the passing of credentials, http/https port, ssl, network modules support an additional top level argument called “provider.” This is supported with both connection local and network_cli `.
`provider` is supported with `connection: local` in < 2.5 and both connection in 2.5.

P.S: please ignore my English, but I hope you got my point.

Reply:
Thanks for the information and example. We are still supporting `provider` with `connection: network_cli` in 2.5, but I thought it was deprecated (see http://docs.ansible.com/ansible/devel/porting_guides/porting_guide_2.5.html?highlight=provider#notice-when-using-provider-dictionary-with-new-persistent-connection-types). For now I've focused on an example for `eapi`. My understanding is that `nxapi` does not support `enable` mode, so I've left that out.